lib/rules/no-extra-boolean-cast.js

```diff
@@ -47,13 +55,40 @@ module.exports = {
             "ForStatement"
         ];
 
+        /**
+         * Check if a node is inside a Boolean function or constructor.
+         * @param {ASTNode} node the node
+         * @returns {boolean} If the node is in the context
+         */
+        function isInFunctionOrConstructorContext(node) {
+            return (node.type === "CallExpression" || node.type === "NewExpression") &&
+                    node.callee.type === "Identifier" &&
+                        node.callee.name === "Boolean";
+
+
+        }
+
+        /**
+         * Checks whether the node is a logical expression and that the option is enabled
+         * @param {ASTNode} node the node
+         * @returns {boolean} if the node is a logical expression and option is enabled
+         */
+        function isInLogicalContext(node) {
+            return node.type === "LogicalExpression" &&
```

Comment:
It would be good to already explicitly make sure that the operator is `&&` or `||`.

The actual estree proposal is to add nullish coalescing `??` as an operator of `LogicalExpression` and `babel-eslint` already works like that. I believe that the "extra boolean cast" logic doesn't apply to `??` in the same way as it applies to `&&` or `||` (though such code probably wouldn't make any sense).